pkg/volume/aws_ebs/aws_util.go

```diff
@@ -214,6 +214,26 @@ func attachDiskAndVerify(b *awsElasticBlockStoreMounter, xvdBeforeSet sets.Strin
 		}
 	}
 
+	glog.V(5).Infof("Failed to attach volume %s: %v", b.volumeID, attachError)
+
+	if awsCloud != nil {
+		detached, err := awsCloud.IsDiskDetached(b.volumeID)
+		if err == nil && !detached {
```

Comment:
hmm.. do you need to do the check here or can you just issue the detach request anyway ? If the previous Attach call is going to eventually succeed then it may not have succeeded yet by the time you did this check.
I wonder how AWS would handle a detach request of the device is detach ?.. does it just bounce or does it get queue behind other requests like the attach request in this case